libs/taskNew.js

```diff
@@ -172,26 +185,48 @@ module.exports = {
                             try {
                                 const body = JSON.parse(data);
                                 fs.unlink(bodyFile, (err) => {
-                                    if (err) cb(err);
-                                    else cb(null, body);
+                                    if (err) cb(err)
+                                    else cb(null, body)
                                 });
                             } catch (e) {
                                 cb(new Error("Malformed body.json"));
                             }
                         }
                     });
                 },
-                (cb) => fs.readdir(srcPath, cb),
+                (cb) => {
+                    fs.readFile(imagesFile, "utf8", (err, data) => {
+                        if (err) cb(err);
+                        else {
+                            try {
+                                const imageLinks = data.split('\n');
+                                imageLinks.pop(); // in order to manipulate data so that it does not contain final \n character
+                                (function downloadImage(imagePath) {
+                                    if (!imagePath) {
+                                        fs.unlink(imagesFile, (err) => {
+                                            if (err) cb(err)
+                                            else cb(null)
+                                        });
+                                    } else {
+                                        const imageName = imagePath.split('/').pop();
+                                        s3.downloadPath(imagePath, `${srcPath}/${imageName}`, (err) => {
+                                            if (err) cb(err)
+                                            else downloadImage(imageLinks.shift())
+                                        })
+                                    }
+                                })(imageLinks.shift())
+                            } catch (e) {
+                                cb(new Error("Malformed images.txt file"));
+                            }
+                        }
+                    });
+                }
             ],
-            (err, [body, files]) => {
+            (err, [body]) => {
                 if (err) res.json({ error: err.message });
                 else {
                     req.body = body;
-                    req.files = files;
 
```

Comment:
these lines should not be deleted. instead, req.files should be [...filesExisting, ...filesDownloaded] (merged file lists)